core/python3Action/lib/launcher.py

```diff
@@ -45,6 +45,26 @@
 # now import the action as process input/output
 from main__ import main as main
 
+class Context:
```

Comment:
Note: It somewhat sucks that we've straight up copied this code over. I'm not too fuzzed about it though given that deduplicating would require us to create a second file and that's a bit of a pain because of how we copy code files around in the non-prelaunched case.

As such, I'd keep this duplicated personally and rather work on getting rid of `launcher.py` wholesale by working through it's technical debt (logshipping).